File: snippets/screens/flutter/maintenance_screen/3/maintenance_screen.dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

class MaintenanceScreen extends StatefulWidget {
  final String? title;
  final String? message;
  final DateTime? estimatedEnd;
  final VoidCallback? onRefresh;
  final String? supportEmail;

  const MaintenanceScreen({
    super.key,
    this.title,
    this.message,
    this.estimatedEnd,
    this.onRefresh,
    this.supportEmail,
  });

  @override
  State<MaintenanceScreen> createState() => _MaintenanceScreenState();
}

class _MaintenanceScreenState extends State<MaintenanceScreen>
    with TickerProviderStateMixin {
  late AnimationController _animationController;
  late AnimationController _toolsController;
  late Animation<double> _fadeAnimation;
  late Animation<double> _slideAnimation;
  late Animation<double> _toolsAnimation;

  @override
  void initState() {
    super.initState();

    SystemChrome.setSystemUIOverlayStyle(
      const SystemUiOverlayStyle(
        statusBarColor: Colors.transparent,
        statusBarIconBrightness: Brightness.dark,
      ),
    );

    _animationController = AnimationController(
      duration: const Duration(milliseconds: 1000),
      vsync: this,
    );

    _toolsController = AnimationController(
      duration: const Duration(milliseconds: 2000),
      vsync: this,
    );

    _fadeAnimation = Tween<double>(
      begin: 0.0,
      end: 1.0,
    ).animate(CurvedAnimation(
      parent: _animationController,
      curve: Curves.easeIn,
    ));

    _slideAnimation = Tween<double>(
      begin: 1.0,
      end: 0.0,
    ).animate(CurvedAnimation(
      parent: _animationController,
      curve: Curves.easeOutCubic,
    ));

    _toolsAnimation = Tween<double>(
      begin: 0.0,
      end: 1.0,
    ).animate(CurvedAnimation(
      parent: _toolsController,
      curve: Curves.easeInOut,
    ));

    _animationController.forward();
    _toolsController.repeat(reverse: true);
  }

  @override
  void dispose() {
    _animationController.dispose();
    _toolsController.dispose();
    super.dispose();
  }

  String _formatEstimatedTime() {
    if (widget.estimatedEnd == null) return '';

    final now = DateTime.now();
    final difference = widget.estimatedEnd!.difference(now);

    if (difference.isNegative) {
      return 'Expected to be resolved soon';
    }

    final hours = difference.inHours;
    final minutes = difference.inMinutes % 60;

    if (hours > 0) {
      return 'Expected back in ${hours}h ${minutes}m';
    } else {
      return 'Expected back in ${minutes}m';
    }
  }

  void _copyEmail() {
    if (widget.supportEmail != null) {
      Clipboard.setData(ClipboardData(text: widget.supportEmail!));
      ScaffoldMessenger.of(context).showSnackBar(
        const SnackBar(
          content: Text('Email copied to clipboard'),
          duration: Duration(seconds: 2),
        ),
      );
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.white,
      body: SafeArea(
        child: AnimatedBuilder(
          animation: _animationController,
          builder: (context, child) {
            return Transform.translate(
              offset: Offset(0, 50 * _slideAnimation.value),
              child: FadeTransition(
                opacity: _fadeAnimation,
                child: Padding(
                  padding: const EdgeInsets.all(24.0),
                  child: Column(
                    children: [
                      const SizedBox(height: 60),
                      // Maintenance illustration
                      AnimatedBuilder(
                        animation: _toolsController,
                        builder: (context, child) {
                          return Container(
                            width: 160,
                            height: 160,
                            decoration: BoxDecoration(
                              color: const Color(0xFFF59E0B).withOpacity(0.1),
                              borderRadius: BorderRadius.circular(80),
                            ),
                            child: Stack(
                              alignment: Alignment.center,
                              children: [
                                // Main wrench icon
                                Icon(
                                  Icons.build,
                                  size: 60,
                                  color: const Color(0xFFF59E0B),
                                ),
                                // Animated tools around
                                Positioned(
                                  top: 20 + (10 * _toolsAnimation.value),
                                  right: 30,
                                  child: Transform.rotate(
                                    angle: _toolsAnimation.value * 0.5,
                                    child: Icon(
                                      Icons.settings,
                                      size: 20,
                                      color: const Color(0xFFF59E0B).withOpacity(0.6),
                                    ),
                                  ),
                                ),
                                Positioned(
                                  bottom: 30 + (8 * _toolsAnimation.value),
                                  left: 25,
                                  child: Transform.rotate(
                                    angle: -_toolsAnimation.value * 0.3,
                                    child: Icon(
                                      Icons.handyman,
                                      size: 18,
                                      color: const Color(0xFFF59E0B).withOpacity(0.5),
                                    ),
                                  ),
                                ),
                              ],
                            ),
                          );
                        },
                      ),
                      const SizedBox(height: 40),
                      // Content
                      Text(
                        widget.title ?? 'Under Maintenance',
                        textAlign: TextAlign.center,
                        style: const TextStyle(
                          fontSize: 28,
                          fontWeight: FontWeight.bold,
                          color: Color(0xFF1F2937),
                        ),
                      ),
                      const SizedBox(height: 16),
                      Text(
                        widget.message ??
                            'We\'re currently performing scheduled maintenance to improve your AIQ experience. Thank you for your patience!',
                        textAlign: TextAlign.center,
                        style: const TextStyle(
                          fontSize: 16,
                          color: Color(0xFF6B7280),
                          height: 1.5,
                        ),
                      ),
                      if (widget.estimatedEnd != null) ...[
                        const SizedBox(height: 24),
                        Container(
                          padding: const EdgeInsets.all(16),
                          decoration: BoxDecoration(
                            color: const Color(0xFFFEF3C7),
                            borderRadius: BorderRadius.circular(12),
                            border: Border.all(
                              color: const Color(0xFFF59E0B).withOpacity(0.3),
                            ),
                          ),
                          child: Row(
                            children: [
                              const Icon(
                                Icons.schedule,
                                color: Color(0xFFF59E0B),
                                size: 20,
                              ),
                              const SizedBox(width: 12),
                              Expanded(
                                child: Text(
                                  _formatEstimatedTime(),
                                  style: const TextStyle(
                                    fontSize: 14,
                                    color: Color(0xFF92400E),
                                    fontWeight: FontWeight.w500,
                                  ),
                                ),
                              ),
                            ],
                          ),
                        ),
                      ],
                      const Spacer(),
                      // Status updates section
                      Container(
                        padding: const EdgeInsets.all(20),
                        decoration: BoxDecoration(
                          color: const Color(0xFFF9FAFB),
                          borderRadius: BorderRadius.circular(16),
                          border: Border.all(
                            color: const Color(0xFFE5E7EB),
                          ),
                        ),
                        child: Column(
                          children: [
                            Row(
                              children: const [
                                Icon(
                                  Icons.info_outline,
                                  color: Color(0xFF6366F1),
                                  size: 20,
                                ),
                                SizedBox(width: 8),
                                Text(
                                  'What we\'re working on:',
                                  style: TextStyle(
                                    fontSize: 14,
                                    fontWeight: FontWeight.w600,
                                    color: Color(0xFF374151),
                                  ),
                                ),
                              ],
                            ),
                            const SizedBox(height: 12),
                            const Text(
                              '• Improving quiz loading performance\n• Adding new AI challenges\n• Enhancing user experience\n• Server optimization',
                              style: TextStyle(
                                fontSize: 14,
                                color: Color(0xFF6B7280),
                                height: 1.4,
                              ),
                            ),
                          ],
                        ),
                      ),
                      const SizedBox(height: 24),
                      // Action buttons
                      Column(
                        children: [
                          SizedBox(
                            width: double.infinity,
                            height: 48,
                            child: ElevatedButton(
                              onPressed: widget.onRefresh ?? () {
                                // Simulate refresh
                                ScaffoldMessenger.of(context).showSnackBar(
                                  const SnackBar(
                                    content: Text('Checking status...'),
                                    duration: Duration(seconds: 2),
                                  ),
                                );
                              },
                              style: ElevatedButton.styleFrom(
                                backgroundColor: const Color(0xFF6366F1),
                                foregroundColor: Colors.white,
                                shape: RoundedRectangleBorder(
                                  borderRadius: BorderRadius.circular(12),
                                ),
                                elevation: 0,
                              ),
                              child: Row(
                                mainAxisAlignment: MainAxisAlignment.center,
                                children: const [
                                  Icon(Icons.refresh, size: 18),
                                  SizedBox(width: 8),
                                  Text(
                                    'Check Status',
                                    style: TextStyle(
                                      fontSize: 14,
                                      fontWeight: FontWeight.w600,
                                    ),
                                  ),
                                ],
                              ),
                            ),
                          ),
                          if (widget.supportEmail != null) ...[
                            const SizedBox(height: 12),
                            SizedBox(
                              width: double.infinity,
                              height: 48,
                              child: OutlinedButton(
                                onPressed: _copyEmail,
                                style: OutlinedButton.styleFrom(
                                  foregroundColor: const Color(0xFF6B7280),
                                  side: const BorderSide(
                                    color: Color(0xFFE5E7EB),
                                  ),
                                  shape: RoundedRectangleBorder(
                                    borderRadius: BorderRadius.circular(12),
                                  ),
                                ),
                                child: Row(
                                  mainAxisAlignment: MainAxisAlignment.center,
                                  children: [
                                    const Icon(Icons.email, size: 18),
                                    const SizedBox(width: 8),
                                    Text(
                                      'Contact Support',
                                      style: const TextStyle(
                                        fontSize: 14,
                                        fontWeight: FontWeight.w500,
                                      ),
                                    ),
                                  ],
                                ),
                              ),
                            ),
                          ],
                        ],
                      ),
                      const SizedBox(height: 16),
                      Text(
                        'Follow us on social media for live updates',
                        textAlign: TextAlign.center,
                        style: TextStyle(
                          fontSize: 12,
                          color: const Color(0xFF9CA3AF),
                        ),
                      ),
                    ],
                  ),
                ),
              ),
            );
          },
        ),
      ),
    );
  }
}

File: snippets/screens/compose/maintenance_screen/3/maintenance_screen.kt
package com.aiq.screens

import androidx.compose.animation.core.*
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.rotate
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import java.util.*

@Composable
fun MaintenanceScreen(
    title: String? = null,
    message: String? = null,
    estimatedEnd: Date? = null,
    onRefresh: () -> Unit = {},
    supportEmail: String? = null
) {
    var toolsRotation by remember { mutableStateOf(0f) }

    val toolsRotationAnim by animateFloatAsState(
        targetValue = toolsRotation,
        animationSpec = infiniteRepeatable(
            animation = tween(2000, easing = EaseInOut),
            repeatMode = RepeatMode.Reverse
        )
    )

    LaunchedEffect(Unit) {
        toolsRotation = 15f
    }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .padding(24.dp),
        horizontalAlignment = Alignment.CenterHorizontally
    ) {
        Spacer(modifier = Modifier.height(60.dp))

        // Maintenance illustration
        Box(
            modifier = Modifier.size(160.dp),
            contentAlignment = Alignment.Center
        ) {
            Card(
                modifier = Modifier.size(160.dp),
                shape = RoundedCornerShape(80.dp),
                colors = CardDefaults.cardColors(
                    containerColor = Color(0xFFF59E0B).copy(alpha = 0.1f)
                )
            ) {
                Box(
                    modifier = Modifier.fillMaxSize(),
                    contentAlignment = Alignment.Center
                ) {
                    Icon(
                        imageVector = Icons.Default.Build,
                        contentDescription = "Maintenance",
                        tint = Color(0xFFF59E0B),
                        modifier = Modifier.size(60.dp)
                    )

                    // Animated tools
                    Icon(
                        imageVector = Icons.Default.Settings,
                        contentDescription = null,
                        tint = Color(0xFFF59E0B).copy(alpha = 0.6f),
                        modifier = Modifier
                            .size(20.dp)
                            .offset(x = 60.dp, y = (-40).dp)
                            .rotate(toolsRotationAnim)
                    )

                    Icon(
                        imageVector = Icons.Default.Handyman,
                        contentDescription = null,
                        tint = Color(0xFFF59E0B).copy(alpha = 0.5f),
                        modifier = Modifier
                            .size(18.dp)
                            .offset(x = (-50).dp, y = 50.dp)
                            .rotate(-toolsRotationAnim * 0.6f)
                    )
                }
            }
        }

        Spacer(modifier = Modifier.height(40.dp))

        // Content
        Text(
            text = title ?: "Under Maintenance",
            fontSize = 28.sp,
            fontWeight = FontWeight.Bold,
            color = Color(0xFF1F2937),
            textAlign = TextAlign.Center
        )

        Spacer(modifier = Modifier.height(16.dp))

        Text(
            text = message ?: "We're currently performing scheduled maintenance to improve your AIQ experience. Thank you for your patience!",
            fontSize = 16.sp,
            color = Color(0xFF6B7280),
            textAlign = TextAlign.Center,
            lineHeight = 24.sp
        )

        estimatedEnd?.let { end ->
            Spacer(modifier = Modifier.height(24.dp))

            Card(
                shape = RoundedCornerShape(12.dp),
                colors = CardDefaults.cardColors(
                    containerColor = Color(0xFFFEF3C7)
                ),
                border = CardDefaults.outlinedCardBorder().copy(
                    brush = androidx.compose.ui.graphics.SolidColor(
                        Color(0xFFF59E0B).copy(alpha = 0.3f)
                    )
                )
            ) {
                Row(
                    modifier = Modifier.padding(16.dp),
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Icon(
                        imageVector = Icons.Default.Schedule,
                        contentDescription = "Time",
                        tint = Color(0xFFF59E0B),
                        modifier = Modifier.size(20.dp)
                    )

                    Spacer(modifier = Modifier.width(12.dp))

                    Text(
                        text = formatEstimatedTime(end),
                        fontSize = 14.sp,
                        color = Color(0xFF92400E),
                        fontWeight = FontWeight.Medium
                    )
                }
            }
        }

        Spacer(modifier = Modifier.weight(1f))

        // Status updates
        Card(
            modifier = Modifier.fillMaxWidth(),
            shape = RoundedCornerShape(16.dp),
            colors = CardDefaults.cardColors(
                containerColor = Color(0xFFF9FAFB)
            ),
            border = CardDefaults.outlinedCardBorder().copy(
                brush = androidx.compose.ui.graphics.SolidColor(Color(0xFFE5E7EB))
            )
        ) {
            Column(
                modifier = Modifier.padding(20.dp)
            ) {
                Row(
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Icon(
                        imageVector = Icons.Default.Info,
                        contentDescription = "Info",
                        tint = Color(0xFF6366F1),
                        modifier = Modifier.size(20.dp)
                    )

                    Spacer(modifier = Modifier.width(8.dp))

                    Text(
                        text = "What we're working on:",
                        fontSize = 14.sp,
                        fontWeight = FontWeight.SemiBold,
                        color = Color(0xFF374151)
                    )
                }

                Spacer(modifier = Modifier.height(12.dp))

                val updates = listOf(
                    "Improving quiz loading performance",
                    "Adding new AI challenges",
                    "Enhancing user experience",
                    "Server optimization"
                )

                updates.forEach { update ->
                    Row(
                        modifier = Modifier.padding(vertical = 2.dp),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Text(
                            text = "•",
                            color = Color(0xFF6B7280),
                            fontSize = 14.sp
                        )

                        Spacer(modifier = Modifier.width(8.dp))

                        Text(
                            text = update,
                            fontSize = 14.sp,
                            color = Color(0xFF6B7280),
                            lineHeight = 20.sp
                        )
                    }
                }
            }
        }

        Spacer(modifier = Modifier.height(24.dp))

        // Action buttons
        Button(
            onClick = onRefresh,
            modifier = Modifier
                .fillMaxWidth()
                .height(48.dp),
            colors = ButtonDefaults.buttonColors(
                containerColor = Color(0xFF6366F1)
            ),
            shape = RoundedCornerShape(12.dp)
        ) {
            Icon(
                imageVector = Icons.Default.Refresh,
                contentDescription = null,
                modifier = Modifier.size(18.dp)
            )

            Spacer(modifier = Modifier.width(8.dp))

            Text(
                text = "Check Status",
                fontSize = 14.sp,
                fontWeight = FontWeight.SemiBold
            )
        }

        supportEmail?.let {
            Spacer(modifier = Modifier.height(12.dp))

            OutlinedButton(
                onClick = { /* Copy email */ },
                modifier = Modifier
                    .fillMaxWidth()
                    .height(48.dp),
                border = BorderStroke(1.dp, Color(0xFFE5E7EB)),
                shape = RoundedCornerShape(12.dp)
            ) {
                Icon(
                    imageVector = Icons.Default.Email,
                    contentDescription = null,
                    modifier = Modifier.size(18.dp),
                    tint = Color(0xFF6B7280)
                )

                Spacer(modifier = Modifier.width(8.dp))

                Text(
                    text = "Contact Support",
                    fontSize = 14.sp,
                    fontWeight = FontWeight.Medium,
                    color = Color(0xFF6B7280)
                )
            }
        }

        Spacer(modifier = Modifier.height(16.dp))

        Text(
            text = "Follow us on social media for live updates",
            fontSize = 12.sp,
            color = Color(0xFF9CA3AF),
            textAlign = TextAlign.Center
        )
    }
}

private fun formatEstimatedTime(date: Date): String {
    val now = Date()
    val difference = date.time - now.time

    if (difference <= 0) {
        return "Expected to be resolved soon"
    }

    val hours = (difference / (1000 * 60 * 60)).toInt()
    val minutes = ((difference % (1000 * 60 * 60)) / (1000 * 60)).toInt()

    return if (hours > 0) {
        "Expected back in ${hours}h ${minutes}m"
    } else {
        "Expected back in ${minutes}m"
    }
}